### PR TITLE
add visual error on canvas fail

### PR DESCRIPTION
**Pull request by cmkohnen**
Add a visual indicator if for some reason the canvas didn't load properly

Bildschirmfoto vom 2023-09-08 16-41-30

### CODE REVIEW

File: src/main.py [hunk 1]
```diff
@@ -127,6 +127,7 @@ def do_activate(self):
             self.set_clipboard(DataClipboard(self))
             self.set_view_clipboard(ViewClipboard(self))
             ui.set_clipboard_buttons(self)
+            ui.reload_canvas(self)
```
[the remark this answers is not in the document]
Comment (cmkohnen) [reply src/main.py:130]: Previously the function was called during the windows init, this just moves it out so we can later implement the window in Vala

File: src/ui.py [hunk 2]
```diff
@@ -278,3 +279,14 @@ def bind_values_to_object(source, window, ignorelist=None):
         except AttributeError:
             logging.warn(_("No way to apply “{}”").format(key))
     return bindings
+
+
+def reload_canvas(self):
```
[the remark this answers is not in the document]
Comment (cmkohnen) [reply src/ui.py:284]: The reason was, to move Canvas instantiation out of the window class to enable porting it to Vala

**Review by Sjoerd1993 — APPROVED** [review]

File: src/ui.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (cmkohnen) [reply src/ui.py:284]: > I was planning to inherit that canvas from the main canvas class, and suspect a reload_canvas function would be useful there. Any specific reason to move this to main?

I have moved the property bindings to the ui function instead, this way the main canvas class could be used more freely. However reloading the canvas can only be implemented in a super function (previously the main window). For A Curve Fitting window, there needs to be a specific implementation for that. Given the move of the bindings we can also restrict that to just show for example the item and the suggested curve for easier visual comparison.

Long story short. This aims to make the canvas Widget a bit more versatile so it could be used idaeally without modification.
